tools/server/server.cpp

```diff
@@ -2569,12 +2577,15 @@ struct server_context {
         res->id        = slot.id_task;
         res->index     = slot.index;
         res->n_tokens  = slot.n_prompt_tokens;
+        res->truncated = slot.truncated;
         res->oaicompat = slot.params.oaicompat;
 
         const int n_embd = llama_model_n_embd(model);
 
         std::vector<float> embd_res(n_embd, 0.0f);
 
+        // Note: If the input was truncated (slot.truncated == true), this embedding
+        // represents only the processed portion of the original input
         for (int i = 0; i < batch.n_tokens; ++i) {
```

Comment:
An easier way to avoid touching this loop - which now becomes more and more fragile - is to simply handle this truncation in `launch_slot_with_task`. You can truncate `slot.prompt_tokens` right after it's being std::moved